File: eCinema/eCinema.Model/Entities/Movie.cs
using eCinema.Models.Enums;

namespace eCinema.Model.Entities
{
    public class Movie
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int DurationMinutes { get; set; }
        public string Language { get; set; }
        public DateTime ReleaseDate { get; set; }
        public MovieStatus Status { get; set; }
        public PgRating PgRating { get; set; }

        public ICollection<MovieGenre> MovieGenres { get; set; } = new List<MovieGenre>();
        public ICollection<MovieActor> MovieActors { get; set; } = new List<MovieActor>();

    }
}

File: eCinema/eCinema.UI/ecinema_mobile/lib/models/movie.dart
import 'package:json_annotation/json_annotation.dart';

part 'movie.g.dart';

@JsonSerializable()
class Movie {
  int? id;
  String? title;
  String? description;
  int? durationMinutes;
  String? language;
  DateTime? releaseDate;
  int? status;
  int? pgRating;
  List<int>? genreIds;
  List<int>? actorIds;
  final String? posterUrl;

  Movie({
    this.id,
    this.title,
    this.description,
    this.durationMinutes,
    this.language,
    this.releaseDate,
    this.status,
    this.pgRating,
    this.genreIds,
    this.actorIds,
    this.posterUrl,
  });

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);
  Map<String, dynamic> toJson() => _$MovieToJson(this);
}
